staging/src/k8s.io/apiextensions-apiserver/test/integration/versioning_test.go

```diff
@@ -136,3 +140,266 @@ func testStoragedVersionInCRDStatus(t *testing.T, ns string, noxuDefinition *api
+	}); err != nil {
+		t.Fatalf("failed to see both v1beta1 and v1beta2 in discovery: %v", err)
+	}
+
+	// ensure that both v1beta1 and v1beta2 are usable by
+	// ensuring that the old CR still has version = v1beta1
+	// and a new CR can be created with version = v1beta2
+	gottenFooInstance, err := noxuV1beta1ResourceClient.Get("foo", metav1.GetOptions{})
+	if err != nil {
+		t.Fatal(err)
+	}
+	if gottenFooInstance.GetAPIVersion() != "mygroup.example.com/v1beta1" {
+		t.Fatalf("invalid version: expected \"mygroup.example.com/v1beta1\", got %v", gottenFooInstance.GetAPIVersion())
+	}
+
+	noxuV1beta2ResourceClient := newNamespacedCustomResourceVersionedClient(ns, dynamicClient, noxuDefinition, "v1beta2")
+	noxuInstanceToCreate = testserver.NewVersionedNoxuInstance(ns, "bar", "v1beta2")
+	createdBarInstance, err := noxuV1beta2ResourceClient.Create(noxuInstanceToCreate)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if createdBarInstance.GetAPIVersion() != "mygroup.example.com/v1beta2" {
+		t.Fatalf("invalid version: expected \"mygroup.example.com/v1beta2\", got %v", createdBarInstance.GetAPIVersion())
+	}
+
+	// set v1beta2 with storage = true
+	gottenCRD, err = apiExtensionClient.ApiextensionsV1beta1().CustomResourceDefinitions().Get(noxuDefinition.Name, metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("could not get CRD: %v", err)
+	}
+	gottenCRD.Spec.Versions[0].Storage = false
+	gottenCRD.Spec.Versions[1].Storage = true
+	err = wait.Poll(500*time.Millisecond, wait.ForeverTestTimeout, func() (bool, error) {
+		updatedCRD, err := apiExtensionClient.ApiextensionsV1beta1().CustomResourceDefinitions().Update(gottenCRD)
+		if err != nil {
+			return false, err
+		}
+
+		if !updatedCRD.Spec.Versions[0].Storage && updatedCRD.Spec.Versions[1].Storage {
+			return true, nil
+		}
+		return false, nil
+	})
+	if err != nil {
+		t.Fatalf("could not update CRD: %v", err)
+	}
+
+	// double check that both v1beta1 and v1beta2 still appear in discovery
+	if err := wait.Poll(100*time.Millisecond, wait.ForeverTestTimeout, func() (done bool, err error) {
+		_, err1 := apiExtensionClient.Discovery().ServerResourcesForGroupVersion(group + "/v1beta1")
+		_, err2 := apiExtensionClient.Discovery().ServerResourcesForGroupVersion(group + "/v1beta2")
+		if err1 == nil && err2 == nil {
+			return true, nil
+		}
+		if errors.IsNotFound(err1) || errors.IsNotFound(err2) {
+			return false, nil
+		}
+		return false, fmt.Errorf("failed to see both v1beta1 and v1beta2 in discovery")
+	}); err != nil {
+		t.Fatalf("failed to see both v1beta1 and v1beta2 in discovery: %v", err)
+	}
+
+	// create a new CR and ensure it persists with version = v1beta2
+	noxuInstanceToCreate = testserver.NewVersionedNoxuInstance(ns, "baz", "v1beta2")
+	createdBazInstance, err := noxuV1beta2ResourceClient.Create(noxuInstanceToCreate)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if createdBazInstance.GetAPIVersion() != "mygroup.example.com/v1beta2" {
```

Comment:
this will not ensure the storage version is v1beta2. Even if it is v1beta1 because you are accessing it using `noxuV1beta2ResourceClient` and hence v1beta2 URL, it will convert it to v1beta2.